tests/run-separate-compilation/f-interpolation-1/FQuote_1.scala

```diff
@@ -0,0 +1,58 @@
+import scala.quoted._
+import scala.tasty.Reflection
+
+import scala.language.implicitConversions
+
+object FQuote {
+
+  implicit class SCOps(ctx: StringContext) {
+    inline def ff(args: => Any*): String = ~impl('(this), '(args))
+  }
+
+  /*private*/ def impl(receiver: Expr[SCOps], args: Expr[Seq[Any]])(implicit reflect: Reflection): Expr[String] = {
+    import reflect._
+
+    def liftListOfAny(lst: List[Term]): Expr[List[Any]] = lst match {
+      case x :: xs  =>
+        val head = x.seal[Any]
+        val tail = liftListOfAny(xs)
+        '{ ~head :: ~tail }
+      case Nil => '(Nil)
+    }
+
+    def isStringConstant(tree: Term) = tree match {
+      case Term.Literal(_) => true
+      case _ => false
+    }
+
+    def isSCOpsConversion(tree: Term) =
+      tree.symbol.fullName == "FQuote$.SCOps"
+
+    def isStringContextApply(tree: Term) =
+      tree.symbol.fullName == "scala.StringContext$.apply"
+
+    // FQuote.SCOps(StringContext.apply([p0, ...]: String*)
+    val parts = receiver.unseal.underlyingArgument match {
+      case Term.Apply(conv, List(Term.Apply(fun, List(Term.Typed(Term.Repeated(values), _)))))
+          if isSCOpsConversion(conv) &&
+             isStringContextApply(fun) &&
+             values.forall(isStringConstant) =>
+        values.collect { case Term.Literal(Constant.String(value)) => value }
+      case tree =>
+        throw new QuoteError(s"String literal expected, but ${tree.show} found")
```

Comment:
For later: better error message with positions.

Reply:
This is just an example for the student to start